### PR TITLE
Refactor/core framework evm

### PR DESCRIPTION
**Pull request by natto1784**
https://app.clickup.com/t/9002019994/PRF-4896
Requires https://github.com/Cypherock/cypherock-common/pull/29

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4896 Ethereum get public key function](https://app.clickup.com/t/9002019994/PRF-4896)

**Comment by natto1784** [comment]
rebased and made the changes

**Review by natto1784 — COMMENTED** [review]
already fixed in https://github.com/Cypherock/x1_wallet_firmware/pull/372
required for the CI build to pass

**Comment by natto1784** [comment]
rebased after https://github.com/Cypherock/x1_wallet_firmware/pull/373

File: apps/evm_family/evm_helpers.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+bool evm_derivation_path_guard(const uint32_t *path, uint32_t depth) {
+  /* types of derivations
+   *
+   * legacy        : m/44'/60'/0'/x
+   * bip44         : m/44'/60'/0'/0/x
+   * account model : m/44'/60'/x'/0/0
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_helpers.c:154]: this comment is fine here but more importantly, the information can go into the doxygen comment as well.

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]

**Pull request merged**